### PR TITLE
handle item to big exceptions DE-12631

### PR DESCRIPTION
**Pull request by slpsys**

### CODE REVIEW

**Comment by dollschasingmen** [comment]
great job marc! amazing! i approve!

**Review by dollschasingmen — APPROVED** [review]

**Comment by slpsys** [comment]
haha, it's still your commit

File: src/main/java/dynamok/sink/DynamoDbSinkTask.java [hunk 1]
```diff
@@ -89,14 +89,25 @@ public void start(Map<String, String> props) {
         remainingRetries = config.maxRetries;
     }
 
+    private void attemptPutItem(SinkRecord record) {
+        try {
+            client.putItem(tableName(record), toPutRequest(record).getItem());
+        } catch (AmazonDynamoDBException e) {
```
Comment (aallawala) [inline src/main/java/dynamok/sink/DynamoDbSinkTask.java:95]: Would there happen to be another exception that could be thrown here from aws that we'd like to catch?

File: src/main/java/dynamok/sink/DynamoDbSinkTask.java [hunk 2]
```diff
@@ -89,14 +89,25 @@ public void start(Map<String, String> props) {
...
+            if(e.getErrorCode().equals("ValidationError")
+                    && e.getErrorMessage().contains("Item size has exceeded the maximum allowed size")) {
+                log.error("Caught item size has exceeded the maximum allowed size exception, item: " + record);
```
Comment (aallawala) [inline src/main/java/dynamok/sink/DynamoDbSinkTask.java]: So if "ValidationError" isn't the error code and error message isn't `Item size...`, this fails silently?

Comment (dollschasingmen) [reply src/main/java/dynamok/sink/DynamoDbSinkTask.java]: hmm -- good point. that is bad, i should re-throw other wise

File: src/main/java/dynamok/sink/DynamoDbSinkTask.java (hunk 1 above)
Comment (dollschasingmen) [reply src/main/java/dynamok/sink/DynamoDbSinkTask.java:95]: not that i know of

Comment (aallawala) [reply src/main/java/dynamok/sink/DynamoDbSinkTask.java:95]: cooooooool cooooooool

Comment (dollschasingmen) [reply src/main/java/dynamok/sink/DynamoDbSinkTask.java:95]: this is our project now so we can add more as they come up

**Review by aallawala — APPROVED** [review]

**Pull request merged**